tools/flang1/flang1exe/lowersym.c

```diff
@@ -5532,7 +5532,7 @@ static void
 llvm_check_retval_inargs(int sptr)
 {
   int fval = FVALG(sptr);
-  if (fval) {
+  if (fval && !IS_PROC_PTR_IFACEG(sptr)) {
```

Comment:
I think the new added code may cannot be covered because all procedure in `ptr_to_internal_proc.f90` is SUBROUTINE and the field FVAL is 0. 
So may be the issue#872 has been fixed by other commit and we only need to revert [82683130](https://github.com/flang-compiler/flang/commit/82683130aeebdbe6d5c70044cd79ef22e54f4f22).

Reply:
@liuxiao0909c you were right! I've updated my commit and removed the overhead change. But the PR itself stays: the new commit not only reverts offending commit but also adds a test case which allows closure of #872.